<a href="https://colab.research.google.com/github/Laura9704/2025_-Analisis/blob/main/TRABAJO.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

librerias

In [11]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

from google.colab import drive
drive.mount('/content/drive')


sns.set_theme(style="whitegrid")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


LLAMADO DE LA DATOS DE RIESGO CARRDIOVASCULAR
INFORMACION OBTENIDA PARA EL TRABAJO DE KAGGLE

In [9]:
data = pd.read_csv('/content/drive/MyDrive/CVD_cleaned.csv')
data.head(10)

General_Health                  Checkup Exercise Heart_Disease Skin_Cancer  \
0           Poor  Within the past 2 years       No            No          No   
1      Very Good     Within the past year       No           Yes          No   
2      Very Good     Within the past year      Yes            No          No   
3           Poor     Within the past year      Yes           Yes          No   
4           Good     Within the past year       No            No          No   
5           Good     Within the past year       No            No          No   
6           Fair     Within the past year      Yes           Yes          No   
7           Good     Within the past year      Yes            No          No   
8           Fair     Within the past year       No            No          No   
9           Fair     Within the past year       No            No          No   

  Other_Cancer Depression Diabetes Arthritis     Sex Age_Category  \
0           No         No       No       Yes  Female        70-74   
1           No         No      Yes        No  Female        70-74   
2           No         No      Yes        No  Female        60-64   
3           No         No      Yes        No    Male        75-79   
4           No         No       No        No    Male          80+   
5           No        Yes       No       Yes    Male        60-64   
6           No         No       No       Yes    Male        60-64   
7           No         No       No       Yes  Female        65-69   
8           No        Yes       No        No  Female        65-69   
9           No         No      Yes       Yes  Female        70-74   

   Height_(cm)  Weight_(kg)    BMI Smoking_History  Alcohol_Consumption  \
0        150.0        32.66  14.54             Yes                  0.0   
1        165.0        77.11  28.29              No                  0.0   
2        163.0        88.45  33.47              No                  4.0   
3        180.0        93.44  28.73              No                  0.0   
4        191.0        88.45  24.37             Yes                  0.0   
5        183.0       154.22  46.11              No                  0.0   
6        175.0        69.85  22.74             Yes                  0.0   
7        165.0       108.86  39.94             Yes                  3.0   
8        163.0        72.57  27.46             Yes                  0.0   
9        163.0        91.63  34.67              No                  0.0   

   Fruit_Consumption  Green_Vegetables_Consumption  FriedPotato_Consumption  
0               30.0                          16.0                     12.0  
1               30.0                           0.0                      4.0  
2               12.0                           3.0                     16.0  
3               30.0                          30.0                      8.0  
4                8.0                           4.0                      0.0  
5               12.0                          12.0                     12.0  
6               16.0                           8.0                      0.0  
7               30.0                           8.0                      8.0  
8               12.0                          12.0                      4.0  
9               12.0                          12.0                      1.0

In [20]:
#tamano del conjunto de datos
data.shape

(308854, 19)

LA BASE DE DATOS TIENE DIESINUEVE CARACTERISTICOS O VARIABLES DE ESTUDIO QUE SE CONSIDERARON PARA EL RIESGO VASCULAR, DE UN 308854 REGISTROS DE INDIVIDUOS.

In [10]:
data.tail()


General_Health                  Checkup Exercise Heart_Disease  \
308849      Very Good     Within the past year      Yes            No   
308850           Fair  Within the past 5 years      Yes            No   
308851      Very Good      5 or more years ago      Yes            No   
308852      Very Good     Within the past year      Yes            No   
308853      Excellent     Within the past year      Yes            No   

       Skin_Cancer Other_Cancer Depression  \
308849          No           No         No   
308850          No           No         No   
308851          No           No        Yes   
308852          No           No         No   
308853          No           No         No   

                                          Diabetes Arthritis     Sex  \
308849                                          No        No    Male   
308850                                         Yes        No    Male   
308851  Yes, but female told only during pregnancy        No  Female   
308852                                          No        No    Male   
308853                                          No        No  Female   

       Age_Category  Height_(cm)  Weight_(kg)    BMI Smoking_History  \
308849        25-29        168.0        81.65  29.05              No   
308850        65-69        180.0        69.85  21.48              No   
308851        30-34        157.0        61.23  24.69             Yes   
308852        65-69        183.0        79.38  23.73              No   
308853        45-49        160.0        81.19  31.71              No   

        Alcohol_Consumption  Fruit_Consumption  Green_Vegetables_Consumption  \
308849                  4.0               30.0                           8.0   
308850                  8.0               15.0                          60.0   
308851                  4.0               40.0                           8.0   
308852                  3.0               30.0                          12.0   
308853                  1.0                5.0                          12.0   

        FriedPotato_Consumption  
308849                      0.0  
308850                      4.0  
308851                      4.0  
308852                      0.0  
308853                      1.0

In [12]:
print("Información del conjunto de datos:\n")
data.info()

Información del conjunto de datos:

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 308854 entries, 0 to 308853
Data columns (total 19 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0   General_Health                308854 non-null  object 
 1   Checkup                       308854 non-null  object 
 2   Exercise                      308854 non-null  object 
 3   Heart_Disease                 308854 non-null  object 
 4   Skin_Cancer                   308854 non-null  object 
 5   Other_Cancer                  308854 non-null  object 
 6   Depression                    308854 non-null  object 
 7   Diabetes                      308854 non-null  object 
 8   Arthritis                     308854 non-null  object 
 9   Sex                           308854 non-null  object 
 10  Age_Category                  308854 non-null  object 
 11  Height_(cm)                   308854 non-null  float64
 12  Weight_(

In [ ]:
print("\n:\n")
data.columns

In [16]:
print("\nEstadísticas descriptivas:")
#data.describe(include="all")
data.describe()


Estadísticas descriptivas:


Height_(cm)    Weight_(kg)            BMI  Alcohol_Consumption  \
count  308854.000000  308854.000000  308854.000000        308854.000000   
mean      170.615249      83.588655      28.626211             5.096366   
std        10.658026      21.343210       6.522323             8.199763   
min        91.000000      24.950000      12.020000             0.000000   
25%       163.000000      68.040000      24.210000             0.000000   
50%       170.000000      81.650000      27.440000             1.000000   
75%       178.000000      95.250000      31.850000             6.000000   
max       241.000000     293.020000      99.330000            30.000000   

       Fruit_Consumption  Green_Vegetables_Consumption  \
count      308854.000000                 308854.000000   
mean           29.835200                     15.110441   
std            24.875735                     14.926238   
min             0.000000                      0.000000   
25%            12.000000                      4.000000   
50%            30.000000                     12.000000   
75%            30.000000                     20.000000   
max           120.000000                    128.000000   

       FriedPotato_Consumption  
count            308854.000000  
mean                  6.296616  
std                   8.582954  
min                   0.000000  
25%                   2.000000  
50%                   4.000000  
75%                   8.000000  
max                 128.000000

In [17]:
#busqueda de datos faltantes
data.isnull().sum()

General_Health                  0
Checkup                         0
Exercise                        0
Heart_Disease                   0
Skin_Cancer                     0
Other_Cancer                    0
Depression                      0
Diabetes                        0
Arthritis                       0
Sex                             0
Age_Category                    0
Height_(cm)                     0
Weight_(kg)                     0
BMI                             0
Smoking_History                 0
Alcohol_Consumption             0
Fruit_Consumption               0
Green_Vegetables_Consumption    0
FriedPotato_Consumption         0
dtype: int64